src/components/datagrid/DataGrid.js

```diff
@@ -60,9 +62,13 @@ export default class DataGridComponent extends NestedArrayComponent {
   get defaultSchema() {
     return DataGridComponent.schema();
   }
-  // Checks if should init the first row (only when no defaultValue and just one row set)
+
+  get initEmpty() {
+    return this.component.initEmpty || this.component.noFirstRow;
+  }
+
   get initRows() {
-    return this.builderMode || !_.isEqual(this.defaultValue, this.emptyValue) || !this.component.noFirstRow;
+    return this.builderMode || this.path === 'defaultValue' || !this.initEmpty;
```

Comment:
Why did you remove the check if the value is empty? This will break the situation where they must now always click "Add Another" if they want to add a new row to a datagrid instead of initializing it with a row open, right?

Reply:
I had to remove it because it appeared to be not a reliable check. Previously I only did it to define the datagrid component that is situated on "Data" tab of datagrid settings where default value is set. 
This check was needed for the situation if you checked an option to hide first row and then wanted to set a default value.
The problem was that you could not properly do it with the rows being hidden.
Now it works such way: if `initEmpty` is set, you can still set default values as normal, but when form is used they will be hidden and only shown gradually when you click "Add Another".

Reply:
So are you saying that this will cause a regression?  When I have a datagrid on my form, it will ALWAYS initialize to no rows and I have to click "Add Another" to add one, unless i have this checkbox checked?  What happens if I do not set the ```noFirstRow``` variable?  I would assume it would always add an initial first row, but this code makes me believe it will still show as empty?  Am I wrong?

Reply:
Here is how it works
https://drive.google.com/file/d/1yczigiLccdsc_9z8NcBMjiMHsACL1zg3/view?usp=sharing

Reply:
noFirstRow option set in JSON will work the same